### ISSUE TITLE
Remove the possibility to change precision for HC pusher in `.param` files.

### ISSUE THREAD

**Issue by Anton-Le** [ComputationalRadiationPhysics/picongpu#3284]
According to an offline discussion with @steindev it is compulsory to use double precision for the computation of the `sqrt` with the HC pusher, introduced with PR#3281.

As such I think it would be better to remove the option to change the precision from `pusher.param`, to prevent usage that is likely to be wrong.

**Comment by sbastrakov** [comment]
So I think the choice is between removing this option from the user side, and hard-set it internally. Which would be a little inconsistent with other pushers, but impossible to misuse. Or to keep it there and `static_assert` that it is set correctly.

P.S. Doesn't the same point apply to the Vay pusher as well?

**Comment by steindev** [comment]
Well, I would think there are cases were one could use the lower precision. I just saw in my examples, that there are cases were you need high precision. I did not check in detail, but I would actually guess that the Boris pusher in the force free test failed miserably was also due to the lower precision. But I did not further investigate as the Boris pusher is not known to produce an erroneous drift velocity in this test. The error could probably orders of magnitude lower with higher precision.

The default in the HC pusher is set to 64bit precision to be on the safe side and to give guidance. And as Sergei already pointed out, since we offer the possibility of reducing the precision in the Vay pusher too, I would leave the implementation as is, even if there is the possibility for misuse. As always, I think we can expect users (which usually belong to the brightest minds as they have a university degree or approach one) to familiarize themselves with the possible side effects when configuring core components and changing defaults. Or at least run small scale tests to see if their setup produces different results after a change.

**Comment by sbastrakov** [comment]
Ah, so there seems to be a difference between your opinions: is double precision required there, or recommended but not strictly required. (I do not have this knowledge currently)

**Comment by steindev** [comment]
> Ah, so there seems to be a difference between your opinions: is double precision required there, or recommended but not strictly required. (I do not have this knowledge currently)

I do not know exactly either. I guess the latter. Maybe *somebody* volunteers to test around a bit? ;-) (But in my opinion we could devote our time to more pressing developments)

**Comment by PrometheusPi** [comment]
I agree with @steindev (and thus disagree with @Anton-Le) the precision should be a user choice. There is a global option to switch to higher precision in PIConGPU as well, but this allows to reduce precision as well - we do not prevent that. The same is true for the fine tuning in pushers and plugins. If a user wants to change the default, he/she is responsible for ensuring that any downsides will not effect his/her simulation or are not relevant. Keeping this option allows to easily run on hardware that has a performance penalty for double precision.

**Comment by steindev** [comment]
@Anton-Le can be closed?

**Comment by psychocoderHPC** [comment]
Ping, is this issue solved?
